node/src/chain/acceptor.rs

```diff
@@ -317,11 +316,104 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
+                                if task.is_running() {
+                                    debug!(
+                                        event = "Rerouting Quorum to Consensus",
+                                        round = qmsg.header.round,
+                                        iter = qmsg.header.iteration,
+                                    );
+                                    task.main_inbound.try_send(msg);
+                                }
+                            }
+                            Err(err) => {
+                                error!("Attestation verification failed: {err}")
+                            }
+                        }
+                    }
+
+                    // If Quorum is for a past round, we only rebroadcast it if
+                    // it's Valid and for a different candidate than what we
+                    // accepted.
+                    //
+                    // The rationale is that:
+                    //  - Fail Quorums have no influence on past rounds
+                    //  - Valid Quorums for accepted candidates have been
+                    //    already broadcast by us
+                    Status::Past => {
+                        let mut discarded = false;
+                        let mut reason = "";
+
+                        if let Vote::Valid(candidate) = qmsg.vote() {
+                            // Check if the candidate is in our chain
+                            if let Ok(candidate_known) = self
+                                .db
+                                .read()
+                                .await
+                                .view(|t| t.get_block_exists(candidate))
+                            {
+                                // If candidate is unknown, rebroadcast
+                                if !candidate_known {
+                                    debug!("Rebroadcast Quorum for past round");
+                                    broadcast(&self.network, &msg).await;
+                                } else {
+                                    discarded = true;
+                                    reason = "past round/iter, or fork";
+                                }
+                            } else {
+                                warn!("Could not check candidate in DB. Skipping Quorum rebroadcast");
+                            };
+                        } else {
+                            discarded = true;
+                            reason = "past round";
+                        }
+
+                        if discarded {
+                            debug!(
+                              event = "Quorum discarded",
+                              reason,
+                              round = qmsg.header.round,
+                              iter = qmsg.header.iteration,
+                              vote = ?qmsg.vote(),
+                            );
+                        }
```

Comment:
Please remove this and use the `debug` statement explicitly where needed. IMO mixing business with logging logic reduces readability